uli_init/systems.py

```diff
@@ -0,0 +1,478 @@
+import json
+import operator
+import os
+import random
+import time
+from collections import namedtuple
+
+import ele
+import foyer
+import gsd
+import hoomd
+import hoomd.md
+import mbuild as mb
+import numpy as np
+import scipy.optimize
+from foyer import Forcefield
+from hoomd.md import wall
+from mbuild.formats.hoomd_simulation import create_hoomd_simulation
+from mbuild.lib.recipes import Polymer
+from scipy.special import gamma
+
+from uli_init.compounds import COMPOUND_DIR
+from uli_init.forcefields import FF_DIR
+from uli_init.utils import base_units
+
+units = base_units.base_units()
+
+
+class System:
+    def __init__(
+        self,
+        molecule,
+        density,
+        system_type,
+        n_compounds=None,
+        polymer_lengths=None,
+        para_weight=None,
+        monomer_sequence=None,
+        forcefield=None,
+        epsilon=1e-7,
+        sample_pdi=False,
+        pdi=None,
+        Mn=None,
+        Mw=None,
+        mass_dist_type="weibull",
+        remove_hydrogens=False,
+        assert_dihedrals=True,
+        seed=24,
+        expand_factor=5
+    ):
+        self.molecule = molecule
+        self.density = density
+        self.type = system_type 
+        self.para_weight = para_weight
+        self.monomer_sequence = monomer_sequence
+        self.forcefield = forcefield
+        self.remove_hydrogens = remove_hydrogens
+        self.assert_dihedrals = assert_dihedrals
+        self.seed = seed
+        self.expand_factor = expand_factor
+        self.target_L = None
+        self.system_mass = 0
+        self.para = 0
+        self.meta = 0
+        self.molecule_sequences = []
+        
+        if self.monomer_sequence and self.para_weight:
+            raise ValueError(
```

Comment:
it might be nice if this was a warning instead. e.g., "you provided both but we're ignoring para_weight. If you want us to use it, don't provide a sequence."

Reply:
Good idea.